src/Workspaces/Core/Portable/Workspace/Solution/TreeAndVersion.cs

```diff
@@ -21,20 +22,13 @@ internal sealed class TreeAndVersion
         /// </summary>
         public VersionStamp Version { get; }
 
-        private TreeAndVersion(SyntaxTree tree, VersionStamp version)
-        {
-            this.Tree = tree;
-            this.Version = version;
-        }
+        public SourceHashAlgorithm ChecksumAlgorithm { get; }
 
-        public static TreeAndVersion Create(SyntaxTree tree, VersionStamp version)
+        public TreeAndVersion(SyntaxTree tree, VersionStamp version, SourceHashAlgorithm checksumAlgorithm)
```

Comment:
Why is checksum algorithm here but not encoding?

Reply:
We get the encoding from `SyntaxTree.Encoding`. Another follow up could be to avoid using that API.

Reply:
https://github.com/dotnet/roslyn/issues/64219